stratum/hal/lib/barefoot/macros.h

```diff
@@ -79,6 +79,14 @@ class BooleanBfStatus {
            << "'" << #expr << "' failed with error message: " \
            << FixMessage(bf_err_str(__ret.status()))
 
+// Same as RETURN_IF_BFRT_ERROR but does not write to LOG(ERROR).
+#define RETURN_IF_BFRT_ERROR_WITHOUT_LOGGING(expr)            \
+  if (const BooleanBfStatus __ret = BooleanBfStatus(expr)) {  \
+  } else /* NOLINT */                                         \
+    return MAKE_ERROR(__ret.error_code()).without_logging()   \
+           << "'" << #expr << "' failed with error message: " \
+           << FixMessage(bf_err_str(__ret.status()))
+
```

Comment:
The duplication here is a little unfortunate, maybe you guys have a better idea.

Reply:
Since we will print the error message in the P4 service, can we just ignore logging by default?

Reply:
I think sometimes the full stack trace is nice to have in the logs.

Reply:
So far I don't have any other idea to reduce the code duplication here